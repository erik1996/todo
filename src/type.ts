export interface IItem {
  id: number;
  title: string;
  status: StatusEnum;
}

export enum StatusEnum {
  TO_DO = "To DO",
  IN_PROGRESS = "In Progress",
  DONE = "Done",
}
