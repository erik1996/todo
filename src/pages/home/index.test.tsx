import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import { HomePage } from "./index";
import renderer from "react-test-renderer";
import { StatusEnum } from "../../type";

describe("HomePage", () => {
  describe("Jest Snapshot testing suite", () => {
    test("renders correctly", () => {
      const component = renderer.create(<HomePage />);
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  test("renders input and add button", () => {
    render(<HomePage />);

    const inputElement = screen.getByPlaceholderText("Add New Item");
    expect(inputElement).toBeInTheDocument();

    const addButton = screen.getByRole("button", { name: "Add" });
    expect(addButton).toBeInTheDocument();
  });

  test("adds item to the list when form is submitted", () => {
    render(<HomePage />);

    const inputElement = screen.getByPlaceholderText("Add New Item");
    const addButton = screen.getByRole("button", { name: "Add" });

    // Enter text in the input field
    fireEvent.change(inputElement, { target: { value: "New Item" } });

    // Submit the form
    fireEvent.click(addButton);

    // Verify that the item is added to the list
    const addedItem = screen.getByText("New Item");
    expect(addedItem).toBeInTheDocument();
  });

  test("removes item from the list when remove button is clicked", () => {
    render(<HomePage />);

    // Add an item to the list
    const inputElement = screen.getByPlaceholderText("Add New Item");
    const addButton = screen.getByRole("button", { name: "Add" });
    fireEvent.change(inputElement, { target: { value: "New Item" } });
    fireEvent.click(addButton);

    // Find the remove button for the added item
    const removeButton = screen.getByRole("button", { name: "Remove" });

    // Click the remove button
    fireEvent.click(removeButton);

    // Verify that the item is removed from the list
    const removedItem = screen.queryByText("New Item");
    expect(removedItem).not.toBeInTheDocument();
  });

  test("updates item status when we select other option", () => {
    render(<HomePage />);
    // Fill the input field and submit the form to add an item
    const inputField = screen.getByPlaceholderText("Add New Item");
    fireEvent.change(inputField, { target: { value: "Test Item" } });
    const addButton = screen.getByText("Add");
    fireEvent.click(addButton);

    // Find the update button and click it
    fireEvent.change(screen.getByLabelText("status"), {
      target: { value: StatusEnum.DONE },
    });

    // Check if the item's status is updated
    const listItem = screen.queryByText(StatusEnum.DONE);

    expect(listItem).toBeInTheDocument();
  });
});
