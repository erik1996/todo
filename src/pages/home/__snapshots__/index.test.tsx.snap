// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage Jest Snapshot testing suite renders correctly 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <input
      name="add-item"
      onChange={[Function]}
      pattern=".{3,20}"
      placeholder="Add New Item "
      required={true}
      title="3 characters minimum and 20 maximum"
      type="text"
      value=""
    />
    <button
      type="submit"
    >
      Add
    </button>
  </form>
  <ul />
</div>
`;
